libs/dal/src/repositories/base-repository.ts

```diff
@@ -77,7 +87,7 @@ export class BaseRepository<T_DBModel, T_MappedEntity, T_Enforcement> {
   }
 
   async *findBatch(
-    query: FilterQuery<T_DBModel> & T_Enforcement,
+    query: QueryData<T_DBModel, T_Enforcement>,
```

Comment:
So my couple of cents related to the issue I initially opened is that in QueryData we do not validate type the same way we do in CreateData, meaning while executing findOne we could pass to it { nonExistingParam: 'id' } which could fail in runtime.
Let me know what you think about it, maybe i missed something on the way.
@rifont